StreamChatCoreTests/Unit Tests/ClientLoggerTests.swift

```diff
@@ -0,0 +1,168 @@
+        
+        allEndpoints = [.guestToken(testUser),
+                        .addDevice(deviceId: "test", testUser),
+                        .devices(testUser),
+                        .removeDevice(deviceId: "test", testUser),
+                        .channels(.init(filter: testFilter)),
+                        .message("test"),
+                        .markAllRead,
+                        .search(.init(filter: testFilter, query: "search", pagination: .none)),
+                        .channel(.init(channel: testChannel, members: testMembers, pagination: .none, options: .all)),
+                        .stopWatching(testChannel),
+                        .updateChannel(.init(data: .init(testChannel))),
+                        .deleteChannel(testChannel),
+                        .hideChannel(testChannel, testUser, true),
+                        .showChannel(testChannel, testUser),
+                        .sendMessage(testMessage, testChannel),
+                        .sendImage("test", "image", testData, testChannel),
+                        .sendFile("test", "image", testData, testChannel),
+                        .deleteImage(testUrl, testChannel),
+                        .deleteFile(testUrl, testChannel),
+                        .markRead(testChannel),
+                        .sendEvent(.channelDeleted, testChannel),
+                        .sendMessageAction(.init(channel: testChannel, message: testMessage, action: .init(name: "test", value: "test", style: .default, type: .button, text: "test"))),
+                        .addMembers(testMembers, testChannel),
+                        .removeMembers(testMembers, testChannel),
+                        .invite(testMembers, testChannel),
+                        .inviteAnswer(.init(channel: testChannel, accept: nil, reject: nil, message: testMessage)),
+                        .replies(testMessage, .none),
+                        .deleteMessage(testMessage),
+                        .addReaction(.angry, testMessage),
+                        .deleteReaction(.angry, testMessage),
+                        .flagMessage(testMessage),
+                        .unflagMessage(testMessage),
+                        .users(.init(filter: testFilter)),
+                        .updateUsers([testUser]),
+                        .muteUser(testUser),
+                        .unmuteUser(testUser),
+                        .flagUser(testUser),
+                        .unflagUser(testUser),
+                        .ban(.init(user: testUser, channel: testChannel, timeoutInMinutes: 0, reason: nil))]
+    }
+    
+    func testLogEndpointStrings() {
+        for endpoint in allEndpoints {
+            logger.log("Endpoint: \(String(describing: endpoint))", level: .debug)
```

Comment:
can't we simply make sure that `Endpoint` can be represented as string safely using Swift? this test smells

Reply:
Yes, we can, but it's no needed :)
There is nothing bad and we use it in our code to log the Endpoint. By default Swift has a good string representation for enums and structs.